homeassistant/components/homematic/__init__.py

```diff
@@ -178,6 +179,7 @@
             vol.Optional(CONF_PATH, default=DEFAULT_PATH): cv.string,
             vol.Optional(CONF_RESOLVENAMES, default=DEFAULT_RESOLVENAMES):
                 vol.In(CONF_RESOLVENAMES_OPTIONS),
+            vol.Optional(CONF_JSONPORT): cv.string,
```

Comment:
Here is where the default port should be set. Not having it here will require everybody to update their configuration.yaml. So you have to define a global `DEFAULT_JSON_PORT` at the top of the file in case there's no global for port 80 set already somewhere else as a constant to use.

Reply:
Oh, wait, maybe I'm wrong here.

Reply:
I have tested it with no "jsonport" in configuration.yaml and it will just not hand it over to pyhomematic which is catching this by leaving out the port from the URL.

Reply:
I thinks technically it's not needed, but for consistency I would still like to ensure that a proper default value is defined.

Reply:
Always sending a default value would make the check in pyhomematic obsolete. So it just always expects the port. In pyhomematic you could then also add the default port. Like it's done with `2001`, which here in the HASS component as well as in pyhomematic always is present as a default value. Think of it as that your change only modifies the default of `80` to something else.

Reply:
Okay, I have added the default for port 80 and adjusted pyhomematic to work accordingly.